src/sonic-yang-models/yang-models/sonic-dhcp-server-ipv4.yang

```diff
@@ -0,0 +1,237 @@
+                    description "Gateway IP for DHCP server";
+                    mandatory true;
+                    type inet:ipv4-address;
+                }
+
+                leaf lease_time {
+                    description "Lease time of DHCP IP in second";
+                    mandatory true;
+                    type uint32 {
+                        range "1..4294967295";
+                    }
+                }
+
+                leaf mode {
+                    description "Mode of assigning IP";
+                    mandatory true;
+                    type string {
+                        pattern "PORT" {
+                            error-message "Invalid mode for DHCP server";
+                            error-app-tag dhcp-server-mode-invalid;
+                        }
+                    }
+                }
+
+                leaf netmask {
+                    description "Subnet mask value for DHCP server";
+                    mandatory true;
+                    type inet:ipv4-address-no-zone;
+                }
+
+                leaf-list customized_options {
+                    description "Customized options list";
+                    type leafref {
+                        path "/dhcp-server-ipv4:sonic-dhcp-server-ipv4/dhcp-server-ipv4:DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS/dhcp-server-ipv4:DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS_LIST/dhcp-server-ipv4:name";
+                    }
+                }
+
+                leaf state {
+                    description "State of DHCP server";
+                    mandatory true;
+                    type enumeration {
+                        enum enabled;
+                        enum disabled;
+                    }
+                }
+            }
+            /* end of DHCP_SERVER_IPV4_LIST */
+        }
+        /* end of DHCP_SERVER_IPV4 container */
+
+        container DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS {
+
+            description "DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS part of config_db.json";
+
+            list DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS_LIST {
+
+                description "DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS list part of config_db.json";
+
+                key "name";
+
+                leaf name {
+                    description "Name of customized option";
+                    type string {
+                        length 1..255 {
+                            error-message "Invalid length for the customized options name";
+                        }
+                    }
+                }
+
+                leaf id {
+                    description "Customized PortChannel0003option ID";
+                    mandatory true;
+                    type uint8 {
+                        range "1..254";
+                    }
+                }
+
+                leaf type {
+                    description "Type of customized option, for standard DHCP option, this field is invalid";
+                    type enumeration {
```

Comment:
Did you test this one?

Reply:
There is `DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS_TYPE_WRONG_VALUE` before to check the invalid value. And I have added `DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS_TYPE_VALID_VALUE_TEXT` etc. to test valid value in newest commit.

Reply:
Do you have test for uint8, uint16 and uint32?